repo2docker/buildpacks/stencila/stencila.py

```diff
@@ -0,0 +1,135 @@
+"""
+Buildpack for stencila editor for DAR document archives
+"""
+
+import os
+import shutil
+
+from ..conda import CondaBuildPack
+
+
+class StencilaBuildPack(CondaBuildPack):
+    """Stencila buildpack
+
+    - installs stencila in /opt/stencila
+    - installs nbserverproxy
+    - registers nbserverproxy extension for stencila
+    """
+
+    @property
+    def runtime(self):
+        """
+        Return contents of runtime.txt if it exists, '' otherwise
+        """
+        if not hasattr(self, "_runtime"):
+            runtime_path = self.binder_path("runtime.txt")
+            try:
+                with open(runtime_path) as f:
+                    self._runtime = f.read().strip()
+            except FileNotFoundError:
+                self._runtime = ""
+
+        return self._runtime
+
+    def detect(self):
+        """
+        Check if current repo should be built with the stencila
+
+        Currently only checks for 'stencila' in runtime.txt
+        """
+        self.manifest_dir = ""
+        for root, dirs, files in os.walk("."):
+            for f in files:
+                if f == "manifest.xml":
+                    self.manifest_dir = os.path.dirname(root)
+                    return True
+        return self.runtime.startswith("stencila")
+
+    def get_build_env(self):
+        """
+        Return build environment variables to be set.
+
+        Sets STENCILA_DIR
+        """
+        return super().get_build_env() + [
+            # This is the path where stencila is installed
+            ("STENCILA_DIR", "/opt/stencila")
+        ]
+
+    def get_env(self):
+        """
+        Return environment variables to be set.
+
+        Sets STENCILA_ARCHIVE_DIR
+        """
+        return super().get_env() + [
+            ("STENCILA_ARCHIVE_DIR", "${HOME}/" + self.manifest_dir)
+        ]
+
+    def get_build_script_files(self):
+        """
+        Dict of files to be copied to the container image for use in building.
+
+        This is copied before the `build_scripts` & `assemble_scripts` are
+        run, so can be executed from either of them.
+
+        It's a dictionary where the key is the source file path in the host
+        system, and the value is the destination file path inside the
+        container image.
+        """
+        return super().get_build_script_files()
```

Comment:
Do we need this if it just calls `super()`?

Reply:
Not at the moment. It used to have things, and it may again before we are done, so I haven't cleaned it out just yet. I'll remove this if it turns out not to be needed in the end.